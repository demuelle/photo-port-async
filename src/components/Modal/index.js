function Modal({photo, toggleModal}) {
    const {name, category, description, index} = photo;
  return (
    <div className="modalBackdrop">
      <div className="modalContainer">
        <h3 data-testid="modal-title" className="modalTitle">{name}</h3>
        <img src={require(`../../assets/large/${category}/${index}.jpg`)} alt={description} />
        <p>{description}</p>
        <button onClick={toggleModal} type="button">Close this modal</button>
      </div>
    </div>
  );
}

export default Modal;
