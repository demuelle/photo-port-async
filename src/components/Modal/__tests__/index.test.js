import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Modal from "..";

const photo = {
  name: "testphoto",
  description: "description of a testphoto",
  category: "food",
  index: 1,
};

const mockToggleModal = jest.fn();

afterEach(cleanup);

describe("Modal component", () => {
  it("renders", () => {
    render(<Modal photo={photo} toggleModal={mockToggleModal} />);
  });

  it("matches snapshot DOM node structure", () => {
    const { asFragment } = render(
      <Modal photo={photo} toggleModal={mockToggleModal} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Modal has a title", () => {
  it("displays the name as the title", () => {
    const { getByTestId } = render(
      <Modal photo={photo} toggleModal={mockToggleModal} />
    );

    expect(getByTestId("modal-title")).toHaveTextContent(photo.name);
  });
});

describe("Click Event", () => {
  it("calls onClose handler", () => {
    // Arrange: Render Modal
    const { getByText } = render(
      <Modal photo={photo} toggleModal={mockToggleModal} />
    );
    // Act: Simulate click event
    fireEvent.click(getByText("Close this modal"));

    // Assert: Expected matcher
    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
});
