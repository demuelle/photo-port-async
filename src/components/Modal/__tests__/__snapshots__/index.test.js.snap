// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
        data-testid="modal-title"
      >
        testphoto
      </h3>
      <img
        alt="description of a testphoto"
        src="1.jpg"
      />
      <p>
        Photo Description
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
